proposals/improved-interpolated-strings.md

```diff
@@ -287,12 +272,106 @@ Do we want to have builders for `System.IFormattable` and `System.FormattableStr
 
 _Answer_: No.
 
-### Lowering
+### Builder pattern codegen
+
+#### `Create` method overload resolution
+
+Given an _applicable\_interpolated\_string\_builder\_type_ `T` and an _interpolated\_string\_expression_ `i`, overload resolution and validation for a valid `Create` method on `T`
+is performed as follows:
+
+1. Member lookup for members with the name `Create` is performed on `T`. The resulting method group is called `M`.
+2. The argument list `A` is constructed as follows:
+    1. The first two arguments are integer constants, representing the literal length of the `i`, and the number of _interpolation_ components in the `i`, respectively.
+    2. If `i` is used as an argument to some parameter `pi` in method `M1`, and parameter `pi` is attributed with `System.Runtime.CompilerServices.InterpolatedStringBuilderArgumentAttribute`,
+    then for every name `Argx` in the `Arguments` array of that attribute the compiler matches it to a parameter `px` that has the same name. The empty string is matched to the receiver
+    of `M1`.
+        * If any `Argx` is not able to be matched to a parameter of `M1`, or an `Argx` requests the receiver of `M1` and `M1` is a static method, an error is produced and no further
+        steps are taken.
+        * Otherwise, the type of every resolved `px` is added to the argument list, in the order specified by the `Arguments` array. These are all passed by value.
+    3. The final argument is a `bool`, passed as an `out` parameter.
+3. Traditional method invocation resolution is performed with method group `M` and argument list `A`.
+    * If a single-best method `F` was found and errors were reported during final validation, those errors are reported and no further steps are taken.
+    * If a single-best method `F` was found and no errors were reported during final validation, the result of overload resolution is `F`.
+    * If no applicable methods were found, step 3 is retried, removing the final `bool` parameter from `A`. If this retry also finds no applicable members, an error is produced and
+    no further steps are taken.
+    * If no single-best method was found, the result of overload resolution is ambiguous, an error is produced, and no further steps are taken.
+4. Final validation on `F` is performed.
+    * If `F`'s return type is not a `T`, an error is produced and no further steps are taken.
+    * If any element of `A` occurred lexically after `i`, an error is produced and no further steps are taken.
+
+By requiring `Create` to be a `static` method instead of a constructor, we allow the implementation to pool builders if it so decides to.
```

Comment:
It looks like the requirement for the method to be static isn't specified anywhere, but this sentence tries to explain the motivation behind the requirement.

Reply:
Traditional method invocation resolution includes making sure that the method is being called from the right context, static in this case.